inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -257,20 +268,35 @@ def raise_if_match_wf_in_error_or_initial(obj, eng):
         None
 
     """
-    def _filter(base_record, match_result):
-        return get_value(
-            match_result, '_source._workflow.status'
-        ) in ('ERROR', 'INITIAL')
-
-    matched_ids = pending_in_holding_pen(obj, _filter)
+    matched_ids = get_wfs_not_completed(obj)
     if bool(matched_ids):
-        obj.extra_data['error_workflows_matched'] = matched_ids
         raise WorkflowsError(
-            'Cannot continue processing. Found workflows in ERROR or INITIAL '
+            'Cannot continue processing.'
+            'Found not completed workflows in holdingpen '
             'state: {}'.format(matched_ids)
         )
 
 
+def remove_blocked_by_me(obj, matched_ids):
```

Comment:
I would do this differently: define a predicate that can then be used to filter the list of matched wfs
```python
def is_workflow_id_blocked_by_workflow(obj, workflow_id):
    wf = workflow_object_class.get(wf_id)
    return obj.id in wf.extra_data.get('holdingpen_matches', [])
```